### PR TITLE
test: use 2GB nondestructive memory and fix failing tests

### PR DESCRIPTION
**Pull request by jkozol**
The cockpit version for testing is bumped to 248. All tests are now run with 2048 mb of nondestructive memory instead of just the image tests. Therefore, the image tests no longer need to explicitly provision more memory.

### CODE REVIEW

**Comment by jkozol** [comment]
@marusak @henrywang I am at a loss for why this test is failing. I have been unable to replicate locally. Do you have any suggestions or thoughts on what could be causing this failure? I have no idea how to continue to debug it.

**Comment by jkozol** [comment]
I would like to merge this PR as is and work in a seperate PR to fix the fedora-33/firefox test failure. These test updates are useful for testing against fedora 34.

File: components/Modal/ManageSources.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
+/* eslint-disable no-unused-vars */
```
Comment (marusak) [inline components/Modal/ManageSources.js:1]: Why is this necessary?

File: components/Modal/ManageSources.js [hunk 2]
```diff
@@ -356,100 +361,109 @@ class ManageSourcesModal extends React.Component {
...
+            description="Sources provide the contents from which components are selected"
+          />
+        )) ||
+          (this.state.editName === "" && <FormattedMessage defaultMessage="Add source" />) || (
+            <FormattedMessage defaultMessage="Edit source" />
+          )}
+      </Title>
+    );
+
+    const body = (
```
Comment (marusak) [inline components/Modal/ManageSources.js:379]: phew, this contains a lot of unnecessary brackets and fragments. It is not directly in return of render, thus you can just write something like:
```
const body = Object.keys(manageSources.sources).length === 0 ?              
          <EmptyState                                                           
            title={formatMessage(messages.errorStateTitle)}                     
            message={formatMessage(messages.errorStateMessage)}                 
          />                                                                    
        : !this.state.addEntry ? 
           <> 
               ...
```

Same for `footer`, But mostly just preference of style, so feel free to ignore :)

File: components/Modal/ManageSources.js [hunk 3]
```diff
@@ -356,100 +361,109 @@ class ManageSourcesModal extends React.Component {
...
+            </button>
+          </>
+        )}
+      </>
+    );
+
+    return (
+      <Modal
+        isOpen
+        variant={ModalVariant.medium}
```
Comment (marusak) [inline components/Modal/ManageSources.js:459]: admittedly, I haven't run this, but we also specify `position="top"`alongside with `variant="medium"`. But it might be, that without this position it aligns more with other existing dialogs?

File: test/verify/check-source [hunk 4]
```diff
@@ -52,17 +52,6 @@ class TestSource(composerlib.ComposerCase):
         # groups action done
         b.click("button:contains('Add source')")
 
-        # HACK: workaround issue https://github.com/osbuild/cockpit-composer/issues/989
```
Comment (marusak) [inline test/verify/check-source:55]: So this commit completely fixes #989 or just one of those problematic dialogs? If all of them, then the commit message could do with `Fixes` line.

**Review by marusak — CHANGES_REQUESTED** [review]
Agree with merging this, no need to block this since current master has that user test broken anyway.

File: test/verify/check-source (hunk 4 above)
Comment (jkozol) [reply test/verify/check-source:55]: Added fixes

File: components/Modal/ManageSources.js (hunk 3 above)
Comment (jkozol) [reply components/Modal/ManageSources.js:449]: Added the `position="top"`

File: components/Modal/ManageSources.js (hunk 2 above)
Comment (jkozol) [reply components/Modal/ManageSources.js]: Refactored for better style

File: components/Modal/ManageSources.js (hunk 1 above)
Comment (jkozol) [reply components/Modal/ManageSources.js]: Thanks for catching this. Removed.

File: components/Modal/UserAccount.js [hunk 5]
```diff
@@ -285,6 +285,7 @@ class UserAccount extends React.Component {
                     <label>
                       <input
                         type="checkbox"
+                        aria-label="Server admin checkbox"
```
[the remark this answers is not in the document]
Comment (jkozol) [reply components/Modal/UserAccount.js]: This is now translated.

**Review by marusak — APPROVED** [review]
Thank you!
